### ISSUE TITLE
What is proper way to reload page with AngularJs 2 and TypeScript?

### ISSUE THREAD

**Issue by newsash** [angular/angular#7076]
`$window.location.reload();` from previous version.

**Comment by zoechi** [comment]
I think I have seen 
```
constructor(private location:Location) {} 
... 
this.location.reload();
```

but not tried myself. 

GitHub issues are for bugs and feature requests. Please use channels like StackOverflow, Gitter, Google groups,... for support questions.

**Comment by vactorlao** [comment]
location.reload();
that work for me

**Comment by zoechi** [comment]
@vactorlao [Location](https://github.com/angular/angular/blob/master/modules/angular2/src/router/location/location.ts#L46) doesn't have a `reload()` method. I don't know where I got this from when I posted my above comment. Curious how it could work for you...

**Comment by DrewynV** [comment]
__________________________________

Hi! We just wanted to let you know we receievd your email. Thank you for submitting your request. Our aim is to get back to you within 24 hours at the latest. If you have any additional information to add to your request, please reply to this email.


Also please visit our knowledgebase (http://www.support.magicdust.com.au) to browse for further info, we are adding to it all the time.

Or here is our industry report about the 11 Essentials for a Successful Website (http://www.magicdust.com.au/web-design/.

We look forward to helping you succeed online.

Thanks

The Team @ Magicdust Digital

http://support.magicdust.com.au/

Ph (02) 9979 7474
M-F | 9 am-5:00 pm AEST (AEDT in Summer) excluding Public Holidays

------------------------------------------------------------------------------------------------------------
This message was sent to  reply@reply.github.com in reference to Case #166564.
------------------------------------------------------------------------------------------------------------


[[f022133a8925e5c47e69245c0201d72dbc23cb83-673074324]]

**Comment by csmrkid** [comment]
@zoechi the "location" is the window.location not ng's location. You may use window.location.reload() or location.reload();

**Comment by nirmalauwucst** [comment]
thanks

**Comment by villanuevadani** [comment]
Some ng-universal-friendly method for this???

**Comment by elijahbalo** [comment]
thanks

**Comment by israellev** [comment]
this is work for me:

`import { Location } from '@angular/common';

constructor(private location: Location) {}

ngOnInit() { load(); }

load() {
  location.reload();
}`

although in the bacillary need to write - this.location.reload();

**Comment by watat83** [comment]
How would you force the reload of the current page from a dialog modal when the dialog is closed?

**Comment by rajnishcoder** [comment]
@israellev  `Property 'reload' does not exist on type 'Location'.` In your code this is Javascript code `location.reload()` so importing location and inject in constructor is waste :-1:

**Comment by predragstevanovic** [comment]
window.location.reload(true);

**Comment by udoyen** [comment]
@predragstevanovic  please where with this be added?

**Comment by haimleibman** [comment]
https://stackoverflow.com/a/46219649

1. Add one more route to same component and navigate to.
2. Or navigate to same route with any params

**Comment by parthkharecha** [comment]
`this.ngOnInit();`
work for me Same route reload.

**Comment by sksandeep087** [comment]
sdfsd

**Comment by parthkharecha** [comment]
In my case i have call resolve + set paramquery for url when all list line check and complete then redirect to other route
Otherwise call same route for call resolve and peremquery get in ngOnInit method.
Hope you are understand.

**Comment by parthkharecha** [comment]
@zoechi other better way https://github.com/angular/angular/issues/7076#issuecomment-381579076

**Comment by shinoybabu** [comment]
DeleteEmployee(id:number)
  {
    this.employeeService.deleteEmployee(id)
    .subscribe( 
      (data) =>{
        console.log(data);
        **this.ngOnInit();**
      }),
      err => {
        console.log("Error occured while deleting the data.");
      }   
  }

**Comment by kaziupir** [comment]
@parthkharecha
It reloads only one component (one where you call onInit).

**Comment by jeserkin** [comment]
To be honest it is a bit funny. Such a redundant thing and is not provided by angular. As far as I have read angular had to be very opinionated framework. How come there is no opinion about that?
Instead users try to invent there own way.
Most popular from this topic looks like `window.location.reload()`. I did head a podcast about how developers who write in angular tend to use VanillaJS instead of angular way. What if angular doesn't provide Angular way?
A bit disappointing.
Sorry for the rant. Probably didn't add anything useful. 😔

**Comment by mlc-mlapis** [comment]
@jeserkin ... why do you need such a feature at all in your case? It's always interesting to know because normally you shouldn't need it.

**Comment by jeserkin** [comment]
@mlc-mlapis I can produce an easy example. Assume you have an application with multiple level rendering.
```
parent
  child 1
  child 2 | parent 2
    child 1 | parent 1
      child 1
      ...
    child 2
    child 3
```

Next, e.g. you have some sort of trigger on `parent`, that needs to reach all of the components somehow.
Sure you can make a service, that would just `.next` new value into subject and pass it around.
Other way would be to set some persistent value cookie, other storage and then read it after page reload with a new state. Other way doesn't require injecting new service in on multiple levels of application.

**tl;dr**
If there is injectable `Location` object, then I don't see why it can't have mechanics to notify router, and behave as if page reload is happening within Angular logic (aka no actual full page reload)

**Comment by mlc-mlapis** [comment]
@jeserkin ... so you are talking about some global interface which all the components would share. Hmm, this way is just not preferable behavior for any part from Angular point of view ... to destroy and re-instantiate the component instances ... and all that only for the reason to say those components that something has been changed. That's why so low attention from the Angular core team ... and probably they are against that logic at all.
